Source/WebCore/Modules/identity/CredentialRequestCoordinator.h

```diff
@@ -27,38 +27,76 @@
 
 #if ENABLE(WEB_AUTHN)
 
-#include "CredentialRequestCoordinatorClient.h"
-#include "IDLTypes.h"
-#include <wtf/CompletionHandler.h>
+#include "BasicCredential.h"
+#include "JSDOMPromiseDeferred.h"
+#include <optional>
+#include <wtf/CanMakeWeakPtr.h>
 #include <wtf/Noncopyable.h>
 #include <wtf/TZoneMalloc.h>
+#include <wtf/UniqueRef.h>
 
 namespace WebCore {
 
 class AbortSignal;
-class BasicCredential;
 class CredentialRequestCoordinatorClient;
 class Document;
-struct CredentialRequestOptions;
-
-template<typename IDLType> class DOMPromiseDeferred;
+class LocalFrame;
+struct DigitalCredentialsRequestData;
+struct DigitalCredentialsResponseData;
+struct ExceptionData;
 
 using CredentialPromise = DOMPromiseDeferred<IDLNullable<IDLInterface<BasicCredential>>>;
 
-class CredentialRequestCoordinator final {
+class CredentialRequestCoordinator final : public RefCounted<CredentialRequestCoordinator>, public CanMakeWeakPtr<CredentialRequestCoordinator> {
     WTF_MAKE_TZONE_ALLOCATED_EXPORT(CredentialRequestCoordinator, WEBCORE_EXPORT);
     WTF_MAKE_NONCOPYABLE(CredentialRequestCoordinator);
 
 public:
-    WEBCORE_EXPORT explicit CredentialRequestCoordinator(std::unique_ptr<CredentialRequestCoordinatorClient>&&);
-    void discoverFromExternalSource(const Document&, CredentialRequestOptions&&, CredentialPromise&&);
+    static Ref<CredentialRequestCoordinator> create(UniqueRef<CredentialRequestCoordinatorClient>&&, Page&);
+    WEBCORE_EXPORT bool presentPicker(CredentialPromise&&, DigitalCredentialsRequestData&&, RefPtr<AbortSignal>);
+    WEBCORE_EXPORT void abortPicker(JSC::JSValue reason);
+
+    ~CredentialRequestCoordinator();
 
 private:
-    CredentialRequestCoordinator() = default;
+    class PickerStateGuard {
+    public:
+        explicit PickerStateGuard(CredentialRequestCoordinator&);
+        PickerStateGuard(const PickerStateGuard&) = delete;
+        PickerStateGuard& operator=(const PickerStateGuard&) = delete;
+
+        PickerStateGuard(PickerStateGuard&&) noexcept;
+        PickerStateGuard& operator=(PickerStateGuard&&) noexcept;
+
+        ~PickerStateGuard();
+
+    private:
+        WeakRef<CredentialRequestCoordinator> m_coordinator;
+        bool m_active { true };
+    }; // class PickerStateGuard
+
+    enum class PickerState : uint8_t {
+        Idle,
+        Presenting,
+        Aborting
+    }; // enum class PickerState
```

Comment:
unnecessary comment

Reply:
I want to keep these comments... it's a bit confusing without it in a code editor I found.